lua/telescope/actions/init.lua

```diff
@@ -805,6 +807,47 @@ actions.cycle_previewers_prev = function(prompt_bufnr)
   actions.get_current_picker(prompt_bufnr):cycle_previewers(-1)
 end
 
+-- keep entry maker
+-- only instantiate new finder
+actions.refine = function(prompt_bufnr, opts)
+  opts = opts or {}
+  opts.prompt_to_prefix = utils.get_default(opts.prompt_to_prefix, true)
+  opts.prefix_hl_group = utils.get_default(opts.prompt_hl_group, "TelescopePromptPrefix")
+  opts.prompt_prefix = utils.get_default(opts.promt_prefix, config.values.prompt_prefix)
+  opts.reset_multi_selection = utils.get_default(opts.reset_multi_selection, false)
+  opts.reset_prompt = utils.get_default(opts.reset_prompt, true)
+  opts.sorter = utils.get_default(opts.sorter, config.values.generic_sorter {})
+
+  local current_picker = action_state.get_current_picker(prompt_bufnr)
+  local entry_maker = current_picker.finder.entry_maker
+
+  local results = {}
+  action_utils.map_entries(prompt_bufnr, function(entry)
+    table.insert(results, entry.value)
+  end)
+
+  -- if opts.sorter == false, keep older sorter
+  if opts.sorter then
+    current_picker.sorter:destroy()
+    current_picker.sorter = opts.sorter
+    current_picker.sorter:_init()
+  end
+
+  local new_finder = finders.new_table { results = results, entry_maker = entry_maker }
+
+  if not opts.reset_multi_selection then
+    opts.multi = current_picker._multi
+  end
+
+  if opts.prompt_to_prefix then
```

Comment:
This could be useful

```lua
if opts.prompt_to_prefix and action_state.get_current_line() ~= "" then
---
```

Reply:
Yes, good point :) I was just always entering somthing in the prompt in my tests 😅